src/main/java/algorithm/VisualizationManager.java

```diff
@@ -0,0 +1,395 @@
+        root.setScaleY(MODEL_SCALE_FACTOR);
+        root.setScaleZ(MODEL_SCALE_FACTOR);
+
+        return root;
+    }
+
+    /**
+     * Creates a new 3d SubScene
+     *
+     * @param root Group containing the nodes Model and Tracker
+     * @return 3d SubScene
+     */
+    private SubScene createScene3D(Group root) {
+        SubScene scene3d = new SubScene(root, VIEWPORT_SIZE, VIEWPORT_SIZE, true, SceneAntialiasing.BALANCED);
+        scene3d.widthProperty().bind(((AnchorPane) meshGroup.getParent()).widthProperty());
+        scene3d.heightProperty().bind(((AnchorPane) meshGroup.getParent()).heightProperty());
+        PerspectiveCamera perspectiveCamera = initCamera();
+
+        scene3d.setFill(Color.DARKGREY);
+        scene3d.setCamera(perspectiveCamera);
+        scene3d.setPickOnBounds(true);
+        return scene3d;
+    }
+
+    /**
+     * Initiates a new PerspectiveCamera
+     *
+     * @return PerspectiveCamera
+     */
+    private PerspectiveCamera initCamera() {
+        PerspectiveCamera perspectiveCamera = new PerspectiveCamera(false);
+        perspectiveCamera.setTranslateX(0);
+        perspectiveCamera.setTranslateY(0);
+        perspectiveCamera.setTranslateZ(0);
+        perspectiveCamera.setNearClip(0.1);
+        perspectiveCamera.setFarClip(1000.0);
+
+        perspectiveCamera.getTransforms().addAll(rotateX, rotateY, new Translate(0, 0, -1000));
+        return perspectiveCamera;
+    }
+
+    /**
+     * Adds Mouse Controls to the Scene
+     *
+     * @param subScene Scene for MouseEvent
+     * @param root     Group containing the nodes to be controlled
+     */
+    private void handleMouse(SubScene subScene, Group root) {
+        subScene.setOnScroll(event -> {
+            double zoomFactor = 1.05;
+            double deltaY = event.getDeltaY();
+            if (deltaY < 0) {
+                zoomFactor = 2.0 - zoomFactor;
+            }
+            root.setScaleX(root.getScaleX() * zoomFactor);
+            root.setScaleY(root.getScaleY() * zoomFactor);
+            root.setScaleZ(root.getScaleZ() * zoomFactor);
+            event.consume();
+        });
+        subScene.setOnMousePressed(event -> {
+            mouseOldX = event.getSceneX();
+            mouseOldY = event.getSceneY();
+        });
+
+        subScene.setOnMouseDragged(event -> {
+            rotateX.setAngle(rotateX.getAngle() - (event.getSceneY() - mouseOldY));
+            rotateY.setAngle(rotateY.getAngle() + (event.getSceneX() - mouseOldX));
+            mouseOldX = event.getSceneX();
+            mouseOldY = event.getSceneY();
+
+        });
+    }
+
+    /**
+     * Adds Keyboard Controls to the Scene
+     *
+     * @param scrollPane        Pane for KeyEvent
+     * @param perspectiveCamera Camera to be controlled
+     */
+    private void handleKeyboard(ScrollPane scrollPane, Camera perspectiveCamera) {
+
+        scrollPane.setOnKeyPressed(event -> {
+            switch (event.getCode()) {
```

Comment:
In order to compile with JDK 11, we unfortunately need to use the old switch-case style with fall-through and break:
```
case S:
case DOWN:
  // logic
  break;
```

Reply:
reverted to old switch style